cirq/ops/common_channels.py

```diff
@@ -187,3 +189,528 @@ def depolarize(p: float) -> DepolarizingChannel:
+
+    def __eq__(self, other):
+        if not isinstance(other, type(self)):
+            return NotImplemented
+        return self._eq_tuple() == other._eq_tuple()
+
+    def __ne__(self, other):
+        return not self == other
+
+    def __repr__(self) -> str:
+        return 'cirq.phase_flip(p={!r})'.format(self._p)
+
+    def __str__(self) -> str:
+        return 'phase_flip(p={!r})'.format(self._p)
+
+    def _circuit_diagram_info_(
+        self, args: protocols.CircuitDiagramInfoArgs
+    ) -> str:
+        return 'PF({!r})'.format(self._p)
+
+
+def phase_flip(p: float) -> PhaseFlipChannel:
+    """
+    Constructs a channel that flips a qubit's phase with probability p
+
+    This channel evovles a density matrix via
+           \rho -> M_0 \rho M_0^\dagger + M_1 \rho M_1^\dagger
+
+    With
+        M_0 = \sqrt{p} \begin{bmatrix} 1 & 0  \\ 0 & 1 \end{bmatrix}
+        M_1 = \sqrt{1-p} \begin{bmatrix} 1 & 0 \\ 0 & -1 \end{bmatrix}
+
+    Args:
+        p: the probability of a phase flip
+
+    Raises:
+        ValueError if p is not a valid probability.
+    """
+    return PhaseFlipChannel(p)
+
+
+class BitFlipChannel(raw_types.Gate):
+    """Channel to flip a qubit"""
+
+    def __init__(self, p) -> None:
+
+        """
+        This channel flips a qubit phase with probability p
+
+        This channel evovles a density matrix via
+            \rho -> M_0 \rho M_0^\dagger + M_1 \rho M_1^\dagger
+
+        With
+            M_0 = \sqrt{p} \begin{bmatrix} 1 & 0  \\ 0 & 1 \end{bmatrix}
+            M_1 = \sqrt{1-p} \begin{bmatrix} 0 & 1 \\ 1 & -0 \end{bmatrix}
+
+        Args:
+            p: the probability of a bit flip
+
+        Raises:
+            ValueError if p is not a valid probability.
+        """
+
+        def validate_probability(p, p_str):
+            if p < 0:
+                raise ValueError('{} was less than 0.'.format(p_str))
+            elif p > 1:
+                raise ValueError('{} was greater than 1.'.format(p_str))
+            return p
+
+        self._p = validate_probability(p, 'p')
+        self._delegate = AsymmetricDepolarizingChannel(1.0 - p, 0.0, 0.0)
+
+    def _channel_(self) -> Iterable[np.ndarray]:
+        # Return just the I and X pieces.
+        return list(self._delegate._channel_())[:2]
+
+    def _eq_tuple(self):
+        return (BitFlipChannel, self._p)
+
+    def __eq__(self, other):
+        if not isinstance(other, type(self)):
+            return NotImplemented
+        return self._eq_tuple() == other._eq_tuple()
+
+    def __ne__(self, other):
+        return not self == other
+
+    def __repr__(self) -> str:
+        return 'cirq.bit_flip(p={!r})'.format(self._p)
+
+    def __str__(self) -> str:
+        return 'bit_flip(p={!r})'.format(self._p)
+
+    def _circuit_diagram_info_(
+        self, args: protocols.CircuitDiagramInfoArgs
+    ) -> str:
+        return 'BF({!r})'.format(self._p)
+
+
+def bit_flip(p: float) -> BitFlipChannel:
```

Comment:
Same comments as for the phase flip method.

Reply:
Think i've got it fixed ?